scene/gui/text_edit.cpp

```diff
@@ -7275,7 +7275,8 @@ void TextEdit::_update_selection_mode_word() {
 		carets.write[caret_idx].selection.selected_word_origin = beg;
 		set_caret_line(line, false, true, 0, caret_idx);
 		set_caret_column(end, true, caret_idx);
-	} else {
+		moved_word_selection = false;
+	} else if (col != carets[caret_idx].selection.selected_word_origin || line != get_selection_line(caret_idx) || moved_word_selection) {
 		if ((col <= carets[caret_idx].selection.selected_word_origin && line == get_selection_line(caret_idx)) || line < get_selection_line(caret_idx)) {
```

Comment:
Sorry, I should have been more clear about why I added the variable.
It was to recreate previous functionality in the editor and in many other text editors where the caret can be moved to the left of a highlighted word, but only after moving the caret elsewhere first:
![prior behaviour](https://github.com/godotengine/godot/assets/83386224/5a82e70b-c7ac-4ba1-bb75-1cfc83af6b04)
If we remove `moved_word_selection` and handle that case separately as suggested, we lose that functionality:
![removed behaviour](https://github.com/godotengine/godot/assets/83386224/d7b217f9-8cf4-4e84-ab72-db346fd477ef)
But with the logic I proposed, we maintain that functionality and the caret doesn't jump around when double-clicking the left side:
![proposed behaviour](https://github.com/godotengine/godot/assets/83386224/ebca63dc-9dd4-4feb-a0ec-2ab010bc7868)

Reply:
Is it desirable to want the caret on the left side of the word? It doesn't change behavior and when the caret is at any other part of the word, it will jump to the right anyway.
Also it feels inconsistent that double clicking on the left side puts the caret on the right end, but moving the cursor to the left end puts it on the left end.

![godot te select word start inconsistent](https://github.com/godotengine/godot/assets/10054226/c51ad022-1a62-4ded-b608-292c40c2a116)

I was basing my logic on VSCode, where the caret is always on the right side of the word. In a different text editor I tried, it is based on the original click position, not just the left end of the word.
And in my opinion, I think the logic is cleaner if we don't have a class variable.

I guess if we do want the caret on the left we could handle it separately and put it on the left.

Reply:
I just wanted to keep the functionality as it was before. However, since this doesn't add anything new and complicates the code, I'll switch it for the suggested change.